test/suite/test_prepare03.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2018 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# test_prepre03.py
+#    Prepare transaction check post conditions for cursor operations
+
+# Pattern of test script is to invoke cursor operations in prepared
+# transactions to ensure they fail and to commit transaction and repeat
+# same operations to ensure normally they pass.
+class test_prepre03(wttest.WiredTigerTestCase):
+    """
+    Test basic operations
+    """
+    table_name1 = 'test_prepare_cursor'
+    nentries = 10
+
+    scenarios = make_scenarios([
+        ('file-col', dict(tablekind='col',uri='file')),
+        ('file-fix', dict(tablekind='fix',uri='file')),
+        ('file-row', dict(tablekind='row',uri='file')),
+        ('lsm-row', dict(tablekind='row',uri='lsm')),
+        ('table-col', dict(tablekind='col',uri='table')),
+        ('table-fix', dict(tablekind='fix',uri='table')),
+        ('table-row', dict(tablekind='row',uri='table'))
+    ])
+
+    def genkey(self, i):
+        if self.tablekind == 'row':
+            return 'key' + str(i)
+        else:
+            return long(i+1)
+
+    def genvalue(self, i):
+        if self.tablekind == 'fix':
+            return int(i & 0xff)
+        else:
+            return 'value' + str(i)
+
+    def assertCursorHasNoKeyValue(self, cursor):
+        keymsg = '/requires key be set/'
+        valuemsg = '/requires value be set/'
+        self.assertRaisesWithMessage(
+            wiredtiger.WiredTigerError, cursor.get_key, keymsg)
+        self.assertRaisesWithMessage(
+            wiredtiger.WiredTigerError, cursor.get_value, valuemsg)
+
+    def session_create(self, name, args):
```

Comment:
Was this just for your debugging purposes? It doesn't seem to be needed in this test since the create is not expected to fail. I think it can be removed now.